posts/kubernetes-tutorial-1-intro.md

```diff
@@ -0,0 +1,69 @@
+---
+authors:
+- David Elizondo
+tags:
+- Kubernetes
+date: 2021-09-07T12:21:50.000Z
+title: "Kubernetes Tutorial 1 Intro"
+image: 
+---
+
+## Intro 
+
+Devops Engineers are now dealing with large application systems in the cloud. One of the most popular orchestration systems currently being used in production across the board is Kubernetes (K8s) under the covers. I want to share some knowledge I have gathered over the years and break down what k8s does at it's core. Understanding of the K8s core construction will allow your to understand how new features are constructed for k8s and how these basic functions ensure a more stable operating environment. 
+
+Kubernetes is an open source project on [github](https://github.com/kubernetes/kubernetes). They define their project as: 
+> Kubernetes, also known as K8s, is an open source system for managing containerized applications across multiple hosts. It provides basic mechanisms for deployment, maintenance, and scaling of applications.
+
+## Application Deployment
+
+To understand K8s, I believe it's helpful to look at and compare how we deployed production applications in the past and what k8s brings to the table. First lets consider the our first old school setup. At company ABC there is a couple of servers in the basement for hosting our applications. Lets say we size our production servers to match our expected user base and concurrent usage. In our example we have four business applications (Green,Orange,Teal,Blue). Lets say all these applications share a network, database and load balancer with each other. We also have a couple of spare servers allocated in the event of a disaster recovery.
+
+![onprem-deployment](https://github.com/David-Elizondo/blog-usa/blob/master/images/2021/09/onprem.png)
+
+## Simulating a Disaster
+
+Lets say the IT get alerted about an outage of application Blue. An electrical storm has come by that night and shorted the hardware on the server. During the outage IT determines that they will need to move the application to one of the spare servers. 
+
+![onprem-deployment](https://github.com/David-Elizondo/blog-usa/blob/master/images/2021/09/onprem2.png)
+
+During the outage IT will be scrambling to bring one of the spare servers online. Configure access the shared database, network load balancer and allowing firewall rules for the application to run on the new server. Each of these task can be time consuming if not planned for. 
```

Comment:
```suggestion
During this outage, IT will be scrambling to bring one of the spare servers online. They must configure access to the shared database and network load balancer, and allow firewall rules for the application to run on the new server. Each of these tasks can be time-consuming if not planned for ahead of time. 
```